evaluator/src/main/kotlin/PackageRule.kt

```diff
@@ -83,21 +83,21 @@ open class PackageRule(
 
     /**
      * A [RuleMatcher] that checks whether any vulnerability for the [package][pkg] has a
-     * [reference][Vulnerability.references] with a [severity][VulnerabilityReference.severity] that equals or is
-     * greater than [threshold] according to the [scoringSystem] and the belonging [severityComparator].
+     * [reference][Vulnerability.references] with a [score][VulnerabilityReference.score] that equals or is
+     * greater than [threshold] according to the [scoringSystem].
      */
-    fun hasVulnerability(threshold: String, scoringSystem: String, severityComparator: (String, String) -> Boolean) =
+    fun hasVulnerability(threshold: Float, scoringSystem: String) =
```

Comment:
We need the `severityComparator` as a fallback if a vulnerability has only a severity but no score. For example, this is the case when VulnerableCode reports vulnerabilities from https://github.com/advisories.

Reply:
Hmm, I wonder whether we should instead have a dedicated rule matcher for that case that takes a severity threshold instead of a score threshold. What do you think?

Reply:
I just had another idea how to implement this... let me give that a try.

Reply:
I'm now avoiding the need for a fallback by mapping the severity to a qualitative rating for ordinal comparison if there is no numeric score.

Reply:
That works for us. However, of course there is one more issue, the scoring system reported by VulnerableCode in those cases is "cvssv3.1_qr", I think currently that does not match any of the aliases.

Reply:
I've added another commit on top to address this.